src/Morphir/JsonSchema/Backend.elm

```diff
@@ -105,55 +111,161 @@ extractTypes modName definition =
             )
 
 
-mapTypeDefinition : TypeName -> Type.Definition ta -> Maybe ( TypeName, SchemaType )
-mapTypeDefinition typeName definition =
+mapTypeDefinition : ( Path, Name ) -> Type.Definition ta -> Result Error (List ( TypeName, SchemaType ))
+mapTypeDefinition (( path, name ) as qualifiedName) definition =
     case definition of
-        Type.TypeAliasDefinition typeArgs typ ->
+        Type.TypeAliasDefinition _ typ ->
             mapType typ
-                |> Maybe.map (Tuple.pair typeName)
-
-        Type.CustomTypeDefinition typeArgs accessControlledConstructors ->
-            Nothing
-
-
-mapType : Type ta -> Maybe SchemaType
+                |> Result.map (\schemaType -> [ ( mapQualifiedName qualifiedName, schemaType ) ])
+
+        Type.CustomTypeDefinition _ accessControlledCtors ->
+            let
+                oneOfs2 =
+                    accessControlledCtors.value
+                        |> Dict.toList
+                        |> List.map
+                            (\( ctorName, ctorArgs ) ->
+                                let
+                                    ctorNameString =
+                                        ctorName |> Name.toTitleCase
+                                in
+                                if List.isEmpty ctorArgs then
+                                    Ok (Const ctorNameString)
+
+                                else
+                                    (ctorArgs
+                                        |> List.map
+                                            (\tpe ->
+                                                mapType (Tuple.second tpe)
+                                            )
+                                    )
+                                        |> ResultList.keepFirstError
+                                        |> Result.map
+                                            (\schemaType ->
+                                                Array
+                                                    (TupleType
+                                                        (Const (ctorName |> Name.toTitleCase)
+                                                            :: schemaType
+                                                        )
+                                                        ((ctorArgs |> List.length) + 1)
+                                                    )
+                                                    False
+                                            )
+                            )
+                        |> ResultList.keepFirstError
+            in
+            oneOfs2
```

Comment:
You could just inline `oneOfs2` here to make it more readable. The naming isn't great anyway.